### PR TITLE
ENT-7560: Added Modules as a language concept

### PR DESCRIPTION
**Pull request by nickanderson**
Ticket: ENT-7560
Changelog: None

### CODE REVIEW

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7109/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7109/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7109/

File: reference/language-concepts/modules.markdown [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+### Specification
+{% endcomment %}
+
+## Promise Modules
+
+Promise modules allow for the implementation of [*custom* promise types][promise-type-custom], extending the CFEngine Language. They communicate with `cf-agent` using the [*Promise Module Protocol*][promise-type-custom#protocol].
+
+**History:**
+
+* Introduced 3.18.0
```
Comment (olehermanse) [inline reference/language-concepts/modules.markdown:24]: Introduced in 3.17, significant fixes and improvements in 3.18.

**Review by olehermanse — APPROVED** [review]
Looks good so far, just one small comment. Thanks for working on this :)

File: reference/language-concepts/modules.markdown (hunk 1 above)
Comment (nickanderson) [reply reference/language-concepts/modules.markdown:24]: thx, will fix introduced version. if you want to specify what was significant in 3.18 we can add that too.

**Pull request merged**